### PR TITLE
Add a guideline for APIs moved on the prototype chain

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/mdn/browser-compat-data/issues/3463 and https://github.com/mdn/browser-compat-data/issues/7517.

### CODE REVIEW

**Comment by foolip** [comment]
I put this together in a hurry to have *something* for review before meetings, so I'm sure there's something I overlooked.

File: docs/data-guidelines.md [hunk 1]
```diff
@@ -280,3 +281,26 @@ Members of this mixin are available to `HTMLAnchorElement` and `HTMLAreaElement`
...
+
+When attributes and methods are moved between interfaces in specifications and implementations, BCD should make the corresponding change. This guidelines covers what versions to use, and whether to use `partial_implementation` and notes in the resulting compat data.
+
+**When members are moved up the prototype chain**, such as from `AudioContext` to `BaseAudioContext`:
+
+- For the members, use the versions when they were first supported on any concrete interface inheriting from `BaseAudioContext`, even if that is earlier than `BaseAudioContext` itself existed. Use `partial_implementation` and notes for the members if there are concrete interfaces where the member wasn't supported prior to the move.
+- For the interface itself, add a `partial_implementation` entry going back as far as the earliest member, and a simple entry from the time the interface itself was introduced.
+
+The [`SVGGeometryElement`](https://github.com/mdn/browser-compat-data/pull/9479) move is an example of this.
+
```
Comment (ddbeck) [inline docs/data-guidelines.md:299]: Can we rework this section to declare a general principle then illustrate it with an example? I'm 95% sure I know what you're getting at here, but without a statement of the principle, I'm not sure.

Something along these lines:

> ### When  members are moved up the prototype chain
> For the members, use the versions when the member was a part of any concrete interface in the prototype chain, even if it was earlier than the existence of the current interface itself. For the interface, use `partial_implementation` and a note on the interface if there are concrete interfaces where the member was not supported before the move.
>
> For example, for member `x` moved from `AudioContext` to `BaseAudioContext`. The data was recorded like this:
> - `x` was removed from `AudioContext`
> - `x` was added to `BaseAudioContext`, with support from the first version supported on `AudioContext`
> - `BaseAudioContext` was shown as having `partial_implementation` from…
> - etc. etc. etc.

File: docs/data-guidelines.md [hunk 2]
```diff
@@ -280,3 +281,26 @@ Members of this mixin are available to `HTMLAnchorElement` and `HTMLAreaElement`
...
+**When members are moved down the prototype chain**, such as from `Node` to `Element`:
```
Comment (ddbeck) [inline docs/data-guidelines.md:300]: A similar situation here. Something like

> ### When members are moved down the prototype chain
>
> Use the version when the member was first supported on the destination interface, ignoring whether it was inherited from the ancestor interface. No `partial_implementation_ or notes about the move are needed.
>
> For example, …

**Review by ddbeck — COMMENTED** [review]
It's quite good already, @foolip. A couple ideas of how we might improve on this, though. Thank you!

**Comment by jpmedley** [comment]
If I'm digesting this correctly, it moves the data closer to representing things where and developer will see them in actual code. If so, LGTM.

**Review by lucalves — APPROVED** [review]

**Comment by foolip** [comment]
@ddbeck I've reworked this almost completely now, PTAL :)

File: docs/data-guidelines.md [hunk 3]
```diff
@@ -302,3 +302,38 @@ Sometimes it's useful to represent support for specific parameters (also known a
...
 
 This guideline was proposed and adopted in [#10509](https://github.com/mdn/browser-compat-data/pull/10509).
+
+## APIs moved on the prototype chain
+
+[Web IDL interfaces](https://heycam.github.io/webidl/#idl-interface) (and [JavaScript built-in objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects)) form prototype chains, with one type inheriting from another. For example, `AudioContext` inherits from `BaseAudioContext`, and `Element` inherits from `Node`.
+
+Some of these interfaces are [abstract](https://en.wikipedia.org/wiki/Abstract_type) and never have instances, while most are concrete and can be instantiated. For example, `BaseAudioContext` and `Node` are abstract, while `AudioContext` and `Element` are concrete.
+
+When attributes and methods are moved between interfaces in specifications and implementations, BCD should make the corresponding change. This guidelines covers what versions to use, and whether to use `partial_implementation` and notes in the resulting compat data.
```
[the remark this answers is not in the document]
Comment (foolip) [reply docs/data-guidelines.md:312]: In the context of this document this is just one guideline. Like https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines.md#parameters-and-parameter-object-features, it's just pretty complicated and covers many cases, but it's all about things moving on the prototype chain.

**Review by jpmedley — APPROVED** [review]

File: docs/data-guidelines.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply docs/data-guidelines.md:312]: I ws trying to fix the subject-verb agreement problem and chose the wrong thing to fix. PTAL.

[the remark this answers is not in the document]
Comment (foolip) [reply docs/data-guidelines.md:312]: Oh, I didn't notice I had a typo there :)

**Comment by foolip** [comment]
@jpmedley per https://github.com/mdn/browser-compat-data/blob/main/GOVERNANCE.md#peers we should not merge non-data pull requests such as this, only the few people in https://github.com/mdn/browser-compat-data/blob/main/GOVERNANCE.md#list-of-current-owners should merge guideline changes.

**Comment by jpmedley** [comment]
Sorry about that.

**Comment by foolip** [comment]
No worries! @ddbeck I'd still like your review and can fix anything that's not quite right in a follow-up.

**Comment by ddbeck** [comment]
I'm happy with this. Thank you, @foolip!

**Pull request merged**